src/API.jl

```diff
@@ -1161,25 +1175,46 @@ function precompile(ctx::Context; internal_call::Bool=false, kwargs...)
                 end
             end
         end
+        push!(tasks, task)
+    end
+    while !interrupted && any(!istaskdone, tasks)
+        try
+            sleep(0.001)
+        catch err
```

Comment:
What happens with `err` if this is not an `InterruptException`?

Reply:
I had thought it was ok because it's just a `sleep`, but I've added a throw for safety

Reply:
Ah, I see now.